app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -87,6 +94,11 @@ def refresh_with_dynflow
       def associate_lifecycle_environments
         self.lifecycle_environments = Katello::KTEnvironment.all if self.default_capsule?
       end
+
+      def associate_content_host
+        content_host = Katello::System.where(:name => self.name).order("created_at DESC").first
+        self.system_id = content_host.id if content_host
```

Comment:
same comment as above, this should be self.content_host_id  (or just use self.content_host)